src/repr/src/timestamp.rs

```diff
@@ -67,6 +67,103 @@ impl RustType<ProtoTimestamp> for Timestamp {
     }
 }
 
+mod columnar_timestamp {
+    use crate::Timestamp;
+    use columnar::Columnar;
+    use mz_ore::cast::CastFrom;
+
+    /// A newtype wrapper for a vector of `Timestamp` values.
+    #[derive(Clone, Copy, Default, Debug)]
+    pub struct TimestampVec<T>(T);
```

Comment:
Bike-shedding: Since the wrapped type doesn't have to be a `Vec`, I wonder if a more generic name would avoid some confusion. How about `Timestamps`?

Reply:
Works for me!